gym/lib/gym/xcodebuild_fixes/swift_fix.rb

```diff
@@ -59,6 +65,42 @@ def swift_library_fix
         end
       end
 
+      # Returns true if all build settings are configured as 2.3
+      # Note: this probably should check the exact build configuration
+      # at some point
+      # @return true if all build settings are configured as 2.3
+      def is_swift_23
+        project = Gym.project
+        xcodeproj_path = project.is_workspace ? project.path.gsub('xcworkspace', 'xcodeproj') : project.path
+
+        project = Xcodeproj::Project.open(xcodeproj_path)
+
+        # Get array of unique swift versions
+        # rubocop:disable Style/MultilineBlockChain
+        swift_versions = project.objects.select do |object|
+          object.isa == 'XCBuildConfiguration'
+        end.map(&:to_hash).map do |object_hash|
+          object_hash['buildSettings']
+        end.select do |build_settings|
+          build_settings.key?('SWIFT_VERSION')
+        end.map do |build_settings|
+          build_settings['SWIFT_VERSION']
+        end.uniq
+
```

Comment:
Should reenable the rubocop style check after the bit we need to skip is done 👍 

# rubocop:enable Style/MultilineBlockChain